app/src/main/java/space/taran/arknavigator/mvp/model/repo/kind/DocumentKindFactory.kt

```diff
@@ -1,22 +1,19 @@
-package space.taran.arknavigator.mvp.model.repo.extra
+package space.taran.arknavigator.mvp.model.repo.kind
 
 import android.graphics.pdf.PdfRenderer
 import android.os.ParcelFileDescriptor
 import android.util.Log
-import space.taran.arknavigator.mvp.model.repo.index.MetaExtraTag
-import space.taran.arknavigator.mvp.model.repo.index.ResourceMetaExtra
 import space.taran.arknavigator.utils.RESOURCES_INDEX
 import space.taran.arknavigator.utils.extension
 import java.io.FileNotFoundException
 import java.nio.file.Path
 
-object DocumentMetaExtra {
-    val ACCEPTED_EXTENSIONS: Set<String> =
+object DocumentKindFactory : ResourceKindFactory {
+    override val acceptedExtensions: Set<String> =
         setOf("pdf", "txt", "doc", "docx", "odt", "ods", "md")
 
-    fun extract(path: Path): ResourceMetaExtra? {
-        val result = mutableMapOf<MetaExtraTag, String>()
-        if (extension(path) != "pdf") return null
+    override fun extract(path: Path): ResourceKind {
+        if (extension(path) != "pdf") return ResourceKind.Document()
```

Comment:
It was `null` before, why `ResourceKind.Document()` now?

Reply:
Because before that we returned a nullable list of extras, now in any case we need to return a kind, and inside properties can be null.